lib/verify.js

```diff
@@ -73,7 +73,22 @@ module.exports = {
     h1.update(hmac.digest());
     h1 = h1.digest();
     var h2 = crypto.createHmac(hashAlg, secret);
-    h2.update(new Buffer(parsedSignature.params.signature, 'base64'));
+
+    var signatureBase64 = parsedSignature.params.signature;
+    var signatureBuffer;
+    if (Buffer.from && Buffer.from !== Uint8Array.from) {
+      // Node.js 4.5.0 and newer
+      signatureBuffer = Buffer.from(signatureBase64, 'base64');
+    } else {
+      // Node.js <4.5.0 || >=5.0.0 <5.10.0
+      if (typeof signatureBase64 === 'number') {
```

Comment:
If we're going to guard against invalid input, wouldn't it make more sense move this check higher? Perhaps it should also be formulated as an `assert` since that's what's being done elsewhere in this code  e.g. 
```
// ...
var signatureBase64 = parsedSignature.params.signature;
assert.string(signatureBase64, 'signatureBase64');
var signatureBuffer;
// ...
```

Reply:
The assertion is made by `Buffer.from` internally. Although I agree `assert.string` can be used for the fallback rather than the type checking.

> A `TypeError` will be thrown if `string` is not a string.

Source: https://nodejs.org/docs/latest-v10.x/api/buffer.html#buffer_class_method_buffer_from_string_encoding

Reply:
OK, so, IIUC, the rationale for pushing the check down into the fall-back code is that it makes it more obvious that it can be removed when the fall-back code is removed. I guess that makes sense.